source/daplink/interface/swd_host_ca.c

```diff
@@ -0,0 +1,949 @@
+/**
```

Comment:
This file has a lot in common with swd_host.c. Do you think it would be possible to share some of the logic, or should these stay as two separate files?